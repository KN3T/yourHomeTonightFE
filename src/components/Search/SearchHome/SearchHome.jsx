import { SearchOutlined, UserOutlined } from '@ant-design/icons';
import {
  AutoComplete,
  Button,
  Col,
  DatePicker,
  Form,
  Input,
  InputNumber,
  Popover,
  Row,
  Spin,
} from 'antd';
import _ from 'lodash';
import moment from 'moment';
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

import { cityApi } from '../../../api';
import homeBg from '../../../assets/images/homeBg.jpg';
import './index.scss';

const { RangePicker } = DatePicker;

const SearchHome = () => {
  const dateFormat = 'YYYY/MM/DD';
  const navigate = useNavigate();
  const [visible, setVisible] = useState(false);
  const [checkIn, setCheckIn] = useState(moment());
  const [checkOut, setCheckOut] = useState(moment().add(3, 'day'));
  const [date, setDate] = useState([
    moment(),
    moment(moment().add(3, 'day').format(dateFormat)),
  ]);
  const [options, setOptions] = useState([]);
  const [beds, setBeds] = useState(1);
  const [children, setChildren] = useState(1);
  const [adults, setAdults] = useState(1);
  const [cityName, setCityName] = useState('An Giang');
  const [loading, setLoading] = useState(false);

  const handleVisibleChange = (newVisible) => {
    setVisible(newVisible);
  };
  const onFinish = (value) => {
    setCityName(value.city);
    setCheckIn(value.date[0].format('YYYY-MM-DD'));
    setCheckOut(value.date[1].format('YYYY-MM-DD'));
    console.log(cityName);
    console.log();
    console.log();
    console.log(beds);
    console.log(adults);
    console.log(children);
    navigate({
      pathname: '/hotels',
      search: `?city=${cityName}&checkIn=${moment(checkIn).format(
        'X'
      )}&checkOut=${moment(checkOut).format(
        'X'
      )}&beds=${beds}&adults=${adults}&children=${children}`,
    });
  };
  const content = (
    <Form
      labelCol={{
        span: 10,
      }}
      initialValues={{
        beds: beds,
        adults: adults,
        children: children,
      }}
    >
      <Form.Item name="beds" label="Beds">
        <InputNumber onChange={(e) => setBeds(e)} min={1} />
      </Form.Item>
      <Form.Item name="adults" label="Adults">
        <InputNumber onChange={(e) => setAdults(e)} min={1} />
      </Form.Item>
      <Form.Item name="children" label="Children">
        <InputNumber onChange={(e) => setChildren(e)} min={1} />
      </Form.Item>
    </Form>
  );

  const onChange = (value) => {
    setCheckIn(value[0].format('YYYY-MM-DD'));
    setCheckOut(value[1].format('YYYY-MM-DD'));
    // console.log(value);
  };
  const search = _.debounce(async (e) => {
    setLoading(true);
    const response = await cityApi.search(e);
    if (response.data.status === 'success') {
      setOptions(
        response.data.data.map((item) => {
          return {
            label: item.city && item.city,
            value: item.city && item.city,
          };
        })
      );
      setLoading(false);
    }
  }, 300);

  const handleSearch = (value) => {
    if (value && value !== '') {
      search(value);
    } else if (value === '') {
      setCityName('');
      setOptions([]);
    }
  };

  const onSelect = (value) => {
    setCityName(value);
  };

  return (
    <div className="search__home__wrapper">
      <Form
        onFinish={onFinish}
        initialValues={{
          city: cityName,
          beds: beds,
          children: children,
          adults: adults,
          date: date,
        }}
      >
        <Row gutter={[10, 0]}>
          <Col
            xl={{ span: 8 }}
            lg={{ span: 8 }}
            md={{ span: 24 }}
            sm={{ span: 24 }}
            xs={{ span: 24 }}
          >
            <Form.Item name="city">
              <AutoComplete
                dropdownMatchSelectWidth={500}
                options={options}
                onSelect={onSelect}
                onSearch={handleSearch}
              >
                <Input
                  onChange={(e) => {
                    setCityName(e.target.value);
                  }}
                  style={{ paddingRight: '50px' }}
                  className="input"
                  size="large"
                  defaultValue={cityName}
                  loading={loading}
                />
              </AutoComplete>
              <Spin className="spin__search" spinning={loading} />
            </Form.Item>
          </Col>
          <Col
            xl={{ span: 8 }}
            lg={{ span: 8 }}
            md={{ span: 24 }}
            sm={{ span: 24 }}
            xs={{ span: 24 }}
          >
            <Form.Item name="date">
              <RangePicker
                className="input"
                // style={{ maxWidth: '310px' }}
                size="large"
                onChange={(value) => onChange(value)}
                defaultValue={[
                  moment(),
                  moment(moment().add(3, 'day').format(dateFormat)),
                ]}
              />
            </Form.Item>
          </Col>
          <Col
            xl={{ span: 8 }}
            lg={{ span: 8 }}
            md={{ span: 24 }}
            sm={{ span: 24 }}
            xs={{ span: 24 }}
          >
            <Form.Item>
              <div className="popover__guests">
                <Popover
                  className="popover__antd"
                  trigger="click"
                  content={content}
                  visible={visible}
                  onVisibleChange={handleVisibleChange}
                >
                  <Button
                    className="btn"
                    size="large"
                    onClick={handleVisibleChange}
                  >
                    {' '}
                    {beds} Beds, {adults} Adults, {children} Children
                  </Button>
                </Popover>
                <div className="search__btn">
                  <Button
                    className="btn"
                    size="large"
                    htmlType="submit"
                    type="primary"
                  >
                    <SearchOutlined />
                  </Button>
                </div>
              </div>
            </Form.Item>
          </Col>
        </Row>
      </Form>
      <div
        className="ctn bg__image__home"
        style={{ backgroundImage: `url(${homeBg})` }}
      ></div>
    </div>
  );
};

export default SearchHome;
