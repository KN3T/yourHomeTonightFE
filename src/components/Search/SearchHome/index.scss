.search__home__wrapper {
  // width: 50vw;
}
