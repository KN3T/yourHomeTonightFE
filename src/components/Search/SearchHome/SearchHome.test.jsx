import { render } from '@testing-library/react';
import moment from 'moment';
import React from 'react';
import { HashRouter } from 'react-router-dom';
import { describe } from 'vitest';

import SearchHome from './SearchHome';

describe('SearchHome component', () => {
  it('should render a search bar', () => {
    const date = [moment('07-09-2022'), moment(moment('07-12-2022'))];
    const searchBar = render(
      <HashRouter>
        <SearchHome
          visible={false}
          date={date}
          options={[
            { city: 'Da Nang', count_hotel: 1 },
            { city: 'Da Lat', count_hotel: 1 },
          ]}
          childrenData="1"
          adults="1"
          cityName="Da Nang"
          loading={false}
          handleVisibleChange={() => console.log(1)}
          onFinish={() => console.log('On finish')}
          handleSearch={() => console.log('search debounce')}
          onSelect={() => console.log('selected')}
          setAdults={() => console.log('adult = 1')}
          setChildren={() => console.log('children = 1')}
          setCityName={() => console.log('city = da nang')}
          DATE_FORMAT="DD-MM-YYYY"
          setDate={() => console.log('date = 1')}
        />
      </HashRouter>
    );
    expect(searchBar).toMatchSnapshot();
  });
});
