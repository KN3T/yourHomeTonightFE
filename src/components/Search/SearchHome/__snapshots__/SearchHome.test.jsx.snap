// Vitest Snapshot v1

exports[`SearchHome component > should render a search bar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="search__home__wrapper"
      >
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="ant-row"
            style="margin-left: -5px; margin-right: -5px;"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-8 ant-col-xl-8"
              style="padding-left: 5px; padding-right: 5px;"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-select ant-select-in-form-item ant-select-auto-complete ant-select-single ant-select-customize-input ant-select-show-search"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                              aria-autocomplete="list"
                              aria-controls="rc_select_TEST_OR_SSR_list"
                              aria-haspopup="listbox"
                              aria-owns="rc_select_TEST_OR_SSR_list"
                              autocomplete="off"
                              class="ant-input ant-input-lg ant-select-selection-search-input input"
                              id="rc_select_TEST_OR_SSR"
                              placeholder="search.search_placeholder"
                              role="combobox"
                              style="padding-right: 50px;"
                              type="search"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-placeholder"
                          />
                        </div>
                      </div>
                      <div
                        aria-busy="false"
                        aria-live="polite"
                        class="ant-spin spin__search"
                      >
                        <span
                          class="ant-spin-dot ant-spin-dot-spin"
                        >
                          <i
                            class="ant-spin-dot-item"
                          />
                          <i
                            class="ant-spin-dot-item"
                          />
                          <i
                            class="ant-spin-dot-item"
                          />
                          <i
                            class="ant-spin-dot-item"
                          />
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-8 ant-col-xl-8"
              style="padding-left: 5px; padding-right: 5px;"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-picker ant-picker-range ant-picker-large input"
                      >
                        <div
                          class="ant-picker-input ant-picker-input-active"
                        >
                          <input
                            autocomplete="off"
                            id="date"
                            placeholder="Start date"
                            readonly=""
                            size="12"
                            value="07-09-2022"
                          />
                        </div>
                        <div
                          class="ant-picker-range-separator"
                        >
                          <span
                            aria-label="to"
                            class="ant-picker-separator"
                          >
                            <span
                              aria-label="swap-right"
                              class="anticon anticon-swap-right"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="swap-right"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                        <div
                          class="ant-picker-input"
                        >
                          <input
                            autocomplete="off"
                            placeholder="End date"
                            readonly=""
                            size="12"
                            value="07-12-2022"
                          />
                        </div>
                        <div
                          class="ant-picker-active-bar"
                          style="left: 0px; width: 0px; position: absolute;"
                        />
                        <span
                          class="ant-picker-suffix"
                        >
                          <span
                            aria-label="calendar"
                            class="anticon anticon-calendar"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="calendar"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-8 ant-col-xl-8"
              style="padding-left: 5px; padding-right: 5px;"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="popover__guests"
                      >
                        <button
                          class="ant-btn ant-btn-default ant-btn-lg btn popover__antd"
                          type="button"
                        >
                          <span>
                             1 search.adults, 1 search.children
                          </span>
                        </button>
                        <div
                          class="search__btn"
                        >
                          <button
                            class="ant-btn ant-btn-primary ant-btn-lg btn"
                            type="submit"
                          >
                            <span
                              aria-label="search"
                              class="anticon anticon-search"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="search"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="search__home__wrapper"
    >
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="ant-row"
          style="margin-left: -5px; margin-right: -5px;"
        >
          <div
            class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-8 ant-col-xl-8"
            style="padding-left: 5px; padding-right: 5px;"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="ant-select ant-select-in-form-item ant-select-auto-complete ant-select-single ant-select-customize-input ant-select-show-search"
                    >
                      <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                            aria-autocomplete="list"
                            aria-controls="rc_select_TEST_OR_SSR_list"
                            aria-haspopup="listbox"
                            aria-owns="rc_select_TEST_OR_SSR_list"
                            autocomplete="off"
                            class="ant-input ant-input-lg ant-select-selection-search-input input"
                            id="rc_select_TEST_OR_SSR"
                            placeholder="search.search_placeholder"
                            role="combobox"
                            style="padding-right: 50px;"
                            type="search"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-placeholder"
                        />
                      </div>
                    </div>
                    <div
                      aria-busy="false"
                      aria-live="polite"
                      class="ant-spin spin__search"
                    >
                      <span
                        class="ant-spin-dot ant-spin-dot-spin"
                      >
                        <i
                          class="ant-spin-dot-item"
                        />
                        <i
                          class="ant-spin-dot-item"
                        />
                        <i
                          class="ant-spin-dot-item"
                        />
                        <i
                          class="ant-spin-dot-item"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-8 ant-col-xl-8"
            style="padding-left: 5px; padding-right: 5px;"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="ant-picker ant-picker-range ant-picker-large input"
                    >
                      <div
                        class="ant-picker-input ant-picker-input-active"
                      >
                        <input
                          autocomplete="off"
                          id="date"
                          placeholder="Start date"
                          readonly=""
                          size="12"
                          value="07-09-2022"
                        />
                      </div>
                      <div
                        class="ant-picker-range-separator"
                      >
                        <span
                          aria-label="to"
                          class="ant-picker-separator"
                        >
                          <span
                            aria-label="swap-right"
                            class="anticon anticon-swap-right"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="swap-right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="0 0 1024 1024"
                              width="1em"
                            >
                              <path
                                d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                      <div
                        class="ant-picker-input"
                      >
                        <input
                          autocomplete="off"
                          placeholder="End date"
                          readonly=""
                          size="12"
                          value="07-12-2022"
                        />
                      </div>
                      <div
                        class="ant-picker-active-bar"
                        style="left: 0px; width: 0px; position: absolute;"
                      />
                      <span
                        class="ant-picker-suffix"
                      >
                        <span
                          aria-label="calendar"
                          class="anticon anticon-calendar"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="calendar"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-8 ant-col-xl-8"
            style="padding-left: 5px; padding-right: 5px;"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="popover__guests"
                    >
                      <button
                        class="ant-btn ant-btn-default ant-btn-lg btn popover__antd"
                        type="button"
                      >
                        <span>
                           1 search.adults, 1 search.children
                        </span>
                      </button>
                      <div
                        class="search__btn"
                      >
                        <button
                          class="ant-btn ant-btn-primary ant-btn-lg btn"
                          type="submit"
                        >
                          <span
                            aria-label="search"
                            class="anticon anticon-search"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="search"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
